### PR TITLE
Generalise the shell script for running Terraform.

### PR DESCRIPTION
**Pull request by sengi**
It's likely that in the near future we'll have a job which needs to run `terraform plan` from Concourse, so generalise this now to stave off the copypasta.

Also add a file comment describing the script, make it pass any additional args to Terraform and add usage instructions.

And some trivial whitespace cleanup.

Tested: pushed to Concourse with `fly sp -t govuk-test -p eks -c eks.yml -l ../parameters/test/eks.yml` (with update-pipeline stuff commented out locally and branch temporarily set to this branch in params), checked that both steps ran successfully: [terraform-eks-cluster](https://cd.gds-reliability.engineering/teams/govuk-test/pipelines/eks/jobs/terraform-eks-cluster/builds/35), [terraform-eks-cluster-addons](https://cd.gds-reliability.engineering/teams/govuk-test/pipelines/eks/jobs/terraform-eks-cluster-addons/builds/23) Also tested the `plan` case locally.

Diff:

```
jobs:

  job terraform-eks-cluster has changed:
  plan:
  - config:
      run:
-       path: govuk-infrastructure/concourse/tasks/terraform-apply.sh
+       args:
+       - apply
+       path: govuk-infrastructure/concourse/tasks/terraform.sh

  job terraform-eks-cluster-addons has changed:
  plan:
  - config:
      run:
-       path: govuk-infrastructure/concourse/tasks/terraform-apply.sh
+       args:
+       - apply
+       path: govuk-infrastructure/concourse/tasks/terraform.sh
```

### CODE REVIEW

File: concourse/tasks/terraform.sh [hunk 1]
```diff
@@ -21,6 +29,6 @@ cd "${DEPLOYMENT_PATH}"
 
 terraform init -input=false -backend-config "${ENVIRONMENT}.backend"
 
-terraform apply -input=false \
+terraform "$tf_action" -input=false \
   -var-file "../variables/${ENVIRONMENT}/common.tfvars" \
-  -auto-approve
+  -auto-approve "$@"
```
Comment (fredericfran-gds) [inline concourse/tasks/terraform.sh:34]: Will this work for plan? Will you not get an error because of the `-auto-approve` flag not defined?

Comment (sengi) [reply concourse/tasks/terraform.sh]: Good catch, thanks. I should have tested both cases. Fixed.

\<rant> It's so frustrating that Terraform doesn't have a consistent way to disable interactive prompting 😭 \</rant>

Comment (sengi) [reply concourse/tasks/terraform.sh]: Also fixed a couple of shellcheck lints.

**Comment by sengi** [comment]
RFAL

**Review by fredericfran-gds — APPROVED** [review]
thanks for fixing...

**Pull request merged**